src/utils/ProfileUtils.ts

```diff
@@ -180,17 +180,23 @@ export async function populateImportedProfile(
     }
 }
 
-//TODO: Check if instead of returning null/empty strings, there's some errors that should be handled
-export async function readProfileFile(file: string) {
-    let read = '';
+export async function readProfileFile(file: string): Promise<string> {
+    let read: string | undefined;
     if (file.endsWith('.r2x')) {
         read = (await FsProvider.instance.readFile(file)).toString();
     } else if (file.endsWith('.r2z')) {
-        const result: Buffer | null = await ZipProvider.instance.readFile(file, "export.r2x");
-        if (result === null) {
-            return null;
-        }
-        read = result.toString();
+        await ZipProvider.instance.readFile(file, "export.r2x").then(
```

Comment:
Combining `await` and `.then()` is quite confusing so I wouldn't use it unless it for some reason is absolutely necessary for the error handling. If combining the approaches, or even if just switching from `await` to `.then()` is for the error handling's sake, I'd leave a comment explaining why it's so. Otherwise someone like me tries to "fix it" later.